File: Java/src/main/java/Solution701_800/Solution771.java
package Solution701_800;

import java.util.HashSet;

/**
 * @author windows
 */
public class Solution771 {
    public int numJewelsInStones(String jewels, String stones) {
        HashSet<Character> set = new HashSet<>();
        for (int i = 0; i < jewels.length(); i++) {
            set.add(jewels.charAt(i));
        }
        int count = 0;
        for (int i = 0; i < stones.length(); i++) {
            if (set.contains(stones.charAt(i))){
                count++;
            }
        }
        return count;
    }


    public static void main(String[] args) {
        Solution771 solution771 = new Solution771();
    }
}

File: C++/src/Solution701_800/Solution771.cpp
//
// Created by windows on 2023/3/2.
//

#include "Solution771.h"
#include <iostream>
#include <vector>
#include <unordered_set>
#include <unordered_map>
#include <string>
using namespace std;


int numJewelsInStones(string jewels, string stones) {
  unordered_set<char> set(jewels.begin(), jewels.end());
  int count = 0;
  for (const auto &item : stones){
    if (set.count(item) != 0){
      count++;
    }
  }
  return count;
}